### PR TITLE
feat(auth): add basic auth to check if the user is admin

### PR DESCRIPTION
**Pull request by smothiki**
Add basic auth for deis cli to authenticate for admin

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -19,6 +23,32 @@ const (
...
+	r.Host = controllerHost
+	req, err := http.NewRequest("GET", "http://"+controllerHost+"/v2/users?limit=1", nil)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header["Authorization"] = r.Header["Authorization"]
+	req.Header["User-Agent"] = r.Header["User-Agent"]
+	if err != nil {
+		return errors.New("cannot read the request body")
+	}
+	c := createHTTPClient(string(sslVerify) == "true")
+	_, err := c.Do(req)
```
Comment (jackfrancis) [inline handlers/handlers.go]: replace the `:=` with `=` on line 48

**Comment by smothiki** [comment]
>Couldn't this use the controller-go-sdk?

can;t use controller-go-sdk as we have to send deis controller api version in the header for the response

File: handlers/handlers.go [hunk 2]
```diff
@@ -20,6 +24,35 @@ const (
...
+
+func checkAdminAuth(r *http.Request) ([]string, error) {
+	sslVerify, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return []string{}, errors.New("cannot read the request body")
+	}
+	if string(sslVerify) == "skip" {
+		return []string{}, nil
+	}
+	urlc := strings.SplitAfterN(r.Host, ".", 2)[1]
```
Comment (arschles) [inline handlers/handlers.go]: can this get a bounds check?

**Comment by arschles** [comment]
@smothiki LGTM assuming https://github.com/deis/workflow-manager/pull/74#discussion_r68433975 is addressed

**Comment by smothiki** [comment]
Closing this PR in favor of recent discussion on how to implement Auth

**Pull request closed without merging**